tests/test_core.py

```diff
@@ -7387,6 +7387,7 @@ def setUp(self):
 asm2g = make_run("asm2g", compiler=CLANG, emcc_args=["-O2", "-g", "-s", "ASSERTIONS=1", "-s", "SAFE_HEAP=1"])
 asm1i = make_run("asm1i", compiler=CLANG, emcc_args=["-O1", '-s', 'EMTERPRETIFY=1'])
 asm3i = make_run("asm3i", compiler=CLANG, emcc_args=["-O3", '-s', 'EMTERPRETIFY=1'])
+asm2m = make_run("asm2m", compiler=CLANG, emcc_args=["-O2", "--memory-init-file", "0", "-s", "MEM_INIT_METHOD=2"])
```

Comment:
we need to test the combination of this option and ASSERTIONS, to check that the assertions do not fire. Can probably keep it on for the entire test mode, but should also test generating code without it (see e.g. `test_cubescript` which does that for a different mode).

it would also be good to get a test showing the assertion firing when the crc is wrong. in `tests/test_other.py`, can emit a file, modify the string manually, then run it and see the error.